Include/Rocket/Core/StringBase.inl

```diff
@@ -87,7 +87,7 @@ StringBase< T >::StringBase(size_type ROCKET_UNUSED_PARAMETER(max_length), const
 template< typename T >
 StringBase< T >::~StringBase()
 {
-	if (value != (T*)local_buffer)
+	if (!IsUsingLocalBuffer())
```

Comment:
No value added, just complexity and obscures the actual check.  It does give the advantage of 'saying what you mean' though and makes it easier to find other bits like this so thats good.